src/execution_plan/ops/op_merge.c

```diff
@@ -164,16 +192,22 @@ OpBase* NewMergeOp(GraphContext *gc, AST *ast, QueryGraph *qg, ResultSet *result
 
 Record OpMergeConsume(OpBase *opBase) {
     OpMerge *op = (OpMerge*)opBase;
-    OpBase *child = op->op.children[0];
 
+    if(op->matched || op->created) return NULL;
```

Comment:
I think that the `matched` part of this logic is not quite right, since it will cause only one entity to be updated:
```
$ redis-cli GRAPH.QUERY social "MATCH (a) WHERE a.age = 31 RETURN COUNT(a)"
1) 1) 1) "COUNT(a)"
   2) 1) "4.000000"
2) 1) "Query internal execution time: 0.277327 milliseconds"
$ redis-cli GRAPH.QUERY social "MERGE (a {age: 31}) SET a.newprop = 100"
1) (empty list or set)
2) 1) "Properties set: 1"
   2) "Query internal execution time: 0.107706 milliseconds"
```

Reply:
Indeed that's the case, 👍